engine/class_modules/sc_death_knight.cpp

```diff
@@ -4450,6 +4457,7 @@ struct empower_rune_weapon_t : public death_knight_spell_t
     dot_duration = base_tick_time = 0_ms;
 
     cooldown -> duration *= 1.0 + p -> vision_of_perfection_minor_cdr;
+    cooldown -> duration -= timespan_t::from_seconds( abs( p -> spec.empower_rune_weapon_rank_2->effectN(1).base_value() / 1000 ) );
```

Comment:
You can save a few operations by replacing ``.base_value() / 1000`` with ``.time_value()`` and simply adding that value to cooldown -> duration, instead of substracting the absolute value.

Reply:
Oh that's much more straightforward